contracts/Token.sol

```diff
@@ -152,7 +181,33 @@ contract Token is Initializable, IToken, IERC20 {
     return allowances[owner][spender];
   }
 
-  // ------------------------ Public but app functions.
+  /// --- ERC1404 functions.
+
+  function detectTransferRestriction (address owner, address recipient, uint256) public view returns (uint8) {
+    IAccess access = reg.access();
+    if (!access.isActor(owner)) {
+      return ERRC_OWNER_NOT_ACTOR;
+    } else if (!access.isActor(recipient)) {
+      return ERRC_RECIPIENT_NOT_ACTOR;
+    } else if (owner == recipient) {
+      return ERRC_OWNER_SAME_AS_RECIPIENT;
+    } else {
+      return 0;
+    }
+  }
+
+  function messageForTransferRestriction (uint8 errCode) public view returns (string memory) {
```

Comment:
This function converts restriction detection error codes into human-readable messages.